tests/cli/test_remote.py

```diff
@@ -1,7 +1,731 @@
+    def test_with_transforms(self, mock_connection: Connection, persist: bool | None):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        original_persist: bool = False
+        remote = SlurmRemote(
+            mock_connection,
+            alloc=alloc,
+            transforms=transforms,
+            persist=original_persist,
+        )
+        new_transforms = [some_other_transform]
+        transformed = remote.with_transforms(*new_transforms, persist=persist)
+        # NOTE: Feels dumb to do this. Not sure what I should be doing otherwise.
+        assert transformed.connection == remote.connection
+        assert transformed.alloc == remote.alloc
+        assert transformed.transforms == [
+            some_transform,
+            some_other_transform,
+            (
+                transformed.srun_transform_persist
+                if persist
+                else transformed.srun_transform
+            ),
+        ]
+        assert transformed._persist == (remote._persist if persist is None else persist)
+
+    @dont_run_for_real
+    @pytest.mark.parametrize("persist", [True, False])
+    def test_persist(self, mock_connection: Connection, persist: bool):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=persist
+        )
+        persisted = remote.persist()
+
+        # NOTE: Feels dumb to do this. Not sure what I should be doing otherwise.
+        assert persisted.connection == remote.connection
+        assert persisted.alloc == remote.alloc
+        assert persisted.transforms == [
+            some_transform,
+            persisted.srun_transform_persist,
+        ]
+        assert persisted._persist is True
+
+    @dont_run_for_real
+    @disable_internet_access
+    def test_ensure_allocation_persist(self, mock_connection: Connection):
+        alloc = ["--time=00:01:00"]
+        transforms = [some_transform]
+        remote = SlurmRemote(
+            mock_connection, alloc=alloc, transforms=transforms, persist=True
+        )
+
+        # TODO: Not sure if this test has any use at this point..
+        remote.extract = Mock(
+            spec=remote.extract,
+            spec_set=True,
+            return_value=(
+                Mock(spec=invoke.runners.Runner, spec_set=True),
+                {"node_name": "bob", "jobid": "1234"},
+            ),
+        )
+
+        results, runner = remote.ensure_allocation()
+
+        remote.extract.assert_called_once_with(
+            "echo @@@ $(hostname) @@@ && sleep 1000d",
+            patterns={
+                "node_name": "@@@ ([^ ]+) @@@",
+                "jobid": "Submitted batch job ([0-9]+)",
+            },
+            hide=True,
+        )
+        assert results == {"node_name": "bob", "jobid": "1234"}
+        # raise NotImplementedError("TODO: Imporant and potentially complicated test")
```

Comment:
I think we can only test this with a real connection to a slurm cluster

Reply:
We can also test this with a real connection, yes.